ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -197,10 +197,39 @@ private[backend] object H2StorageBackend
       connectionInitHook: Option[Connection => Unit],
   )(implicit loggingContext: LoggingContext): DataSource = {
     val h2DataSource = new org.h2.jdbcx.JdbcDataSource()
-    h2DataSource.setUrl(jdbcUrl)
+
+    // H2 (org.h2.jdbcx.JdbcDataSource) does not support setting the user/password within the jdbcUrl, so remove
+    // those properties from the url if present and set them separately. Note that Postgres and Oracle support
+    // user/password in the URLs, so we don't bother exposing user/password configs separately from the url just for h2
+    // which is anyway not supported for production. (This also helps run canton h2 participants that set user and
+    // password.)
+    val urlNoUser = setKeyValueAndRemoveFromUrl(jdbcUrl, "user", h2DataSource.setUser)
```

Comment:
The parsing logic seems a non-trivial string logic, how about:
- having a more streamlined signature to the pure function like def extractUserPasswordAndRemoveFromUrl(jdbcUrl: String): (String, Option[String], Option[String]) // (clensedUrl, foundUser, foundPassword)
- having some simple unit tests to guard the pure function